pyrsistent/_helpers.py

```diff
@@ -31,7 +31,7 @@ def freeze(o, strict=True):
     (1, pvector([]))
     """
     typ = type(o)
-    if typ is dict or (strict and isinstance(o, PMap)):
+    if isinstance(o, dict) or (strict and isinstance(o, PMap)):
```

Comment:
While this certainly works for what you want to achieve I'm sligthly worried of the unintended effects it may have in clients code bases. Things that may not previously have been froozen (such as `OrderedDict`s , `Counter`s, etc.) now start becoming frozen which leads to data loss and change of existing functionality.

If we want freeze to work also for `defaultdicts` I would prefer adding it as a distinct type to check against (using `is`) given the above considerations. Although not as nice/general I think it would be a safer move with a smaller blast radius. It's still a backwards imcompatible change, but a smaller one.

Also, please add a test for your new case while you are at it. :-)

Reply:
Makes sense! Thank you for the review and suggestions.
I have made the changes accordingly.

Also, I've added tests for ```defaultdict```.